Code/Tools/AWSNativeSDKInit/source/AWSLogSystemInterface.cpp

```diff
@@ -40,15 +44,12 @@ namespace AWSNativeSDKInit
     Aws::Utils::Logging::LogLevel AWSLogSystemInterface::GetLogLevel() const
     {
         Aws::Utils::Logging::LogLevel newLevel = m_logLevel;
-        static const char* const logLevelEnvVar = "sys_SetLogLevel";
-        auto logVar = AZ::Environment::FindVariable<int>(logLevelEnvVar);
-
-        if (logVar)
+        int awsLogLevel = bg_awsLogLevel;
```

Comment:
Querying of this variable needs to be done via the `AZ::IConsole::GetCvarValue` API.

Why you didn't ask? Well I'll answer the question that you didn't ask.

The AWSNativeSDKInit is a static library.
It can be linked in to multiple shared libraries and an executable.
The `AZ_CVAR` macro just declares an [AZ::ConsoleDataWrapper](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Console/IConsole.h#L214-L216) at namespace scope, therefore it is a global variable that has copy per library it is linked into.

The problem this causes is that when a `ConsoleFunctor` is linked into the AZ::Console via [Console::RegisterFunctor](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Console/Console.cpp#L302-L331) function when a module loads.

By default the DontDuplicate flag isn't set for a CVar so multiple CVars with the same name can be linked into the AZ::Console.
When AZ [Console::DispatchCommand](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Console/Console.cpp#L432-L462) is used to invoke a ConsoleCommand, it executes for each functor that is linked that has been registered with the same name. So they all get updated at once.

Now here is where a problem can occur.
The DispatchCommand can **only update the linked CVars that are registered at the time the function is called**.
What this means is if the shared libraries that linked to the AWSNativeSDKInit library are loaded after the `bg_awsLogLevel` has been executed, will not have their global variable instance set at that point.


For example the AWSNativeSDKInit library is linked into the [EditorLib](https://github.com/o3de/o3de/blob/development/Code/Editor/CMakeLists.txt#L113) which loads way before any gems load.
And then the `bg_awsLogLevel` Console Command is invoked before any Gems modules that link to the AWSNativeSDKInit library are loaded such as the [AWSCore.Editor](https://github.com/o3de/o3de/blob/development/Gems/AWSCore/Code/CMakeLists.txt#L27-L87) module.
Then if that module invoked `GetLogLevel` function that value it would return would be the default if the AWSLogSystemInterface was created from it's memory space..

Reply:
Now a potential fix for this issue is when the AZ::Console registers a functor with the same name, it uses GetCVarValue to read to the current value from the already registered functor and sets that into the newly registered functor

Reply:
Synced offline, it is a console system issue, which cvar (defined in static lib) won't be loaded with latest value if trying to get from variable directly. Have to using console interface `GetCVarValue` function actively. Will update this in next revision.